include/oneapi/dpl/pstl/glue_memory_impl.h

```diff
@@ -329,14 +318,13 @@ uninitialized_value_construct(_ExecutionPolicy&& __exec, _ForwardIterator __firs
     {
         oneapi::dpl::__internal::__pattern_walk_brick(
             __dispatch_tag, std::forward<_ExecutionPolicy>(__exec), __first, __last,
-            oneapi::dpl::__internal::__brick_fill<decltype(__dispatch_tag), _DecayedExecutionPolicy, _ValueType>{
-                _ValueType()});
+            oneapi::dpl::__internal::__brick_fill<decltype(__dispatch_tag), _ValueType>{_ValueType()});
     }
     else
     {
         oneapi::dpl::__internal::__pattern_walk1(
             __dispatch_tag, std::forward<_ExecutionPolicy>(__exec), __first, __last,
-            oneapi::dpl::__internal::__op_uninitialized_value_construct<_DecayedExecutionPolicy>{});
+            oneapi::dpl::__internal::__op_uninitialized_value_construct<std::decay_t<_ExecutionPolicy>>{});
```

Comment:
As far as we are changing template parameters of `__op_uninitialized_value_construct`....
It looks `__op_uninitialized_value_construct` has only one implementation for the host and device purposes. So, let remove `_ExecutionPolicy` at all.

Reply:
Fixed

Reply:
I don't know from implementation point of view how to do this.
@MikeDvorskiy could you please prepare some separate PR with this fix?